app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt

```diff
@@ -150,6 +182,52 @@ fun PocketStories(
     }
 }
 
+private fun Modifier.onShown(
```

Comment:
are there any performance concerns with this running a bunch of calculations on the UI thread?

Reply:
Pretty much all calculations will be avoided with the help of `if (!wasEventReported) ...` once the composable was visible. Before that, while the screen is scrolled this `onGloballyPositioned` will fire multiple times but I don't expect any visible performance impact as there are just a few, simple calculations.